data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -136,19 +137,25 @@ public String executeToJsonResult(String database, String command, ClientRequest
         command = command.trim();
         CommandType commandType = determineCommandType(command);
         long timeoutMs = determineTimeout(properties, commandType);
+        // TODO save the uri once - no need to format everytime
         String clusterEndpoint = String.format(commandType.getEndpoint(), clusterUrl);
```

Comment:
Where do we format again?

Reply:
we reach this code for each command executed - this can be done once